static/preview-code.js

```diff
@@ -1,8 +1,16 @@
-import hljs from 'https://cdn.jsdelivr.net/npm/highlightjs@9.16.2/+esm'
+import hljs from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/highlight.min.js';
+import javascript from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/languages/javascript.min.js';
+import yaml from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/languages/yaml.min.js';
+import css from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/languages/css.min.js';
+import json from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/languages/json.min.js';
+hljs.registerLanguage('javascript', javascript);
+hljs.registerLanguage('yaml', yaml);
+hljs.registerLanguage('css', css);
+hljs.registerLanguage('json', json);
 
 const inscription = document.documentElement.dataset.inscription;
 const response = await fetch(`/content/${inscription}`);
+const contentType = response.headers.get("content-type");
+const language = contentType.split("/")[1];
 const code = await response.text();
-const codeBlock = document.createElement('code');
-codeBlock.innerHTML = hljs.highlightAuto(code).value;
-document.body.appendChild(codeBlock);
+document.body.innerHTML = `<pre><code>${hljs.highlight(code, {language, ignoreIllegals: true}).value}</code></pre>`;
```

Comment:
I missed the `<pre>` which was in the screenshot which does the formatting. Refactored it a bit.